### ISSUE TITLE
BaseModel.new method doesn't returns corresponding type hint.

### ISSUE THREAD

**Issue by jcfernandez-890825** [odoo-ide/pycharm-odoo#34]
When BaseModel.new method is used over a specific model ("sale.order", "sale.subscription", etc.), type hint for the returned value shows BaseModel instead of the corresponding Model ("sale.order", "sale.subscription", etc.).
![image](https://user-images.githubusercontent.com/28876647/137719206-b3f2a0ff-ea44-4670-86e6-63ca2437ace6.png)

This doesn't happen with BaseModel.create method.
![image](https://user-images.githubusercontent.com/28876647/137719651-407b8d5f-687c-42b0-b12b-43e61c508343.png)

Could we get the same behavior for BaseModel.new method?

**Comment by trinhanhngoc** [comment]
Hi @jcfernandez-890825 

It's weird. I just tested again, it still works as expected.

**Comment by jcfernandez-890825** [comment]
Hi @trinhanhngoc 

Maybe I'm missing something?
Could you show a print-screen  will the test you did? Or maybe a sort video?

**Comment by trinhanhngoc** [comment]
@jcfernandez-890825 Ok

![Peek 2021-10-18 21-03](https://user-images.githubusercontent.com/11208291/137746902-8954390f-6414-42d5-a614-24aafbc8d6d3.gif)

**Comment by jcfernandez-890825** [comment]
Later I'm going to Invalidate Caches and see if that fix the issue.

**Comment by jcfernandez-890825** [comment]
Here do the following:


```python
class SaleSubscription(models.Model):
    # Private attributes
    _inherit = "sale.subscription"

    # Action methods
    def _create_storable_prod_so(self):
        sale_orders = self.env['sale.order']
        for subscription in self:
            new_lines = self.env['sale.subscription.line']
            for line_dct in subscription.recurring_invoice_line_ids.filtered(
                    'product_id.storable_product_id'
            ).read([
                'product_id', 'quantity'
            ]):
                new_line = new_lines.new(new_lines._convert_to_write(line_dct))
                new_line_prod = new_line.product_id.storable_product_id
                new_line.update({
                    'product_id': new_line_prod,
                    'uom_id': new_line_prod.uom_id,
                    'name': new_line_prod.get_product_multiline_description_sale()
                })
                new_lines |= new_line

            if new_lines:
                so_dct = self._prepare_renewal_order_values(
                    discard_product_ids=subscription.mapped(
                        'recurring_invoice_line_ids.product_id'
                    ).filtered('storable_product_id').ids,
                    new_lines_ids=new_lines
                )[self.id]
                so_dct['subscription_management'] = 'recurring'

                so = sale_orders.create(so_dct)
                so.order_line._compute_tax_id()
                sale_orders |= so

        return sale_orders
```

With the code above, `new_line` type hint is `BaseModel`:
![image](https://user-images.githubusercontent.com/28876647/137761938-768868fd-8f1f-49c9-97f8-5e2f25c4eb41.png)

But if you remove the selected code, `new_line` type hint is `sale.subscription.line`:
![image](https://user-images.githubusercontent.com/28876647/137762444-08560c64-b401-4070-b608-30f9ce24cefd.png)


Any idea on what might be the problem?

**Comment by trinhanhngoc** [comment]
@jcfernandez-890825 I got it. The fix will be available in the next release. Thank you so much!

**Comment by jcfernandez-890825** [comment]
Hi @trinhanhngoc 

As you promised, this issue is fixed on v2021.2.8.
But it will be nice if you could get us the goodies/features we already enjoyed from `BaseModel.mapped` into `BaseModel.filtered`.

;)

**Comment by trinhanhngoc** [comment]
> Hi @trinhanhngoc
> 
> As you promised, this issue is fixed on v2021.2.8. But it will be nice if you could get us the goodies/features we already enjoyed from `BaseModel.mapped` into `BaseModel.filtered`.
> 
> ;)

Yes, It's just implemented a few minutes ago :))
